### ISSUE TITLE
[Bug]: Can't use `unplugin-vue-components/rspack` plugin

### ISSUE THREAD

**Issue by zhylmzr** [web-infra-dev/rsbuild#897]
### Version

```sh
System:
    OS: Windows 11 10.0.22621
    CPU: (8) x64 Intel(R) Core(TM) i7-9700 CPU @ 3.00GHz
    Memory: 14.18 GB / 31.94 GB
  Browsers:
    Edge: Chromium (120.0.2210.61)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @rsbuild/core: ^0.1.9 => 0.1.9 
    @rsbuild/plugin-vue: ^0.1.9 => 0.1.9
```

### Details

`unplugin-vue-components` works fine in rspack, but fails in rsbuild.

```ts
// rsbuild.config.ts
import { defineConfig } from '@rsbuild/core'
import { pluginVue } from '@rsbuild/plugin-vue'

import Component from 'unplugin-vue-components/rspack'
import { AntDesignVueResolver } from 'unplugin-vue-components/resolvers'

export default defineConfig({
  plugins: [pluginVue()],
  tools: {
    rspack: (config, { env }) => {
      config.plugins?.push(
        Component({
          // resolvers: [AntDesignVueResolver({ importStyle: false })],
        })
      )
    },
  },
})
```

The error:

```
start   Use Rspack v0.4.1
start   Compiling...
[unplugin-vue-components] no components found
error   Compile error:
Failed to compile, check the errors for troubleshooting.
error[internal]: [html-rspack-plugin]: Expression expected
Expected ';', '}' or <eof>
C:\...\node_modules\.pnpm\@rsbuild+core@0.1.9\node_modules\@rsbuild\core\static\template.html:31
            throw e;
            ^

Error: Expression expected
Expected ';', '}' or <eof>
```

### Reproduce link

https://github.com/zhylmzr/rsbuild-1

### Reproduce Steps

1. use `pnpm create rsbuild@latest` to create vue3 project with typescript
2. install plugin `pnpm add unplugin-vue-components -D`
3. copy the config to `rsbuild.config.ts`
4. run and failed

**Comment by chenjiahan** [comment]
I think this is still a bug with Rspack, and I have created an issue for Rspack: https://github.com/web-infra-dev/rspack/issues/4963

**Comment by zhylmzr** [comment]
fixed in unplugin 1.6.0
